wasmtime-api/src/runtime.rs

```diff
@@ -79,10 +79,22 @@ impl Engine {
 
 // Store
 
+pub(crate) trait SignatureRegistry {
+    fn register_cranelift_signature(
+        &mut self,
+        signature: &ir::Signature,
+    ) -> wasmtime_runtime::VMSharedSignatureIndex;
+    fn lookup_cranelift_signature(
+        &self,
+        type_index: wasmtime_runtime::VMSharedSignatureIndex,
+    ) -> Option<&ir::Signature>;
+}
```

Comment:
Can you talk briefly about why this is a trait? What other `SignatureRegistry` implementations will there be?

Reply:
Adding comment: "The SignatureRegistry allow register new cranelift signature, and also quickly retrive the signature back using registration index."

But it is very similar to wasmtime_runtime::SignatureRegistry, except it does not allow lookup. (Assuming due to memory/performance reasons?)

Reply:
I'm still unclear on this; the trait is `pub(crate)` and there's only one implementation in the crate. Do you expect there will be additional implementations?

Reply:
Right. I guess I'm planning too aggressively for future uses. I will remove the trait for now.